File: CatalogoFilmes.API/models/movie.cs
using System.ComponentModel.DataAnnotations;

namespace CatalogoFilmes.API.Models
{
    public class Movie
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(200)]
        public string Titulo { get; set; } = string.Empty;
        
        [Range(1900, 2030)]
        public int Ano { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Genero { get; set; } = string.Empty;
        
        [Required]
        [StringLength(150)]
        public string Diretor { get; set; } = string.Empty;
        
        [StringLength(1000)]
        public string Sinopse { get; set; } = string.Empty;
        
        public string? PosterUrl { get; set; }

        [Range(1, 10, ErrorMessage = "A avaliação deve estar entre 1 e 10.")]
        public int Avaliacao { get; set; }


        
        public DateTime DataLancamento { get; set; }
        
        [Range(1, 500)]
        public int Duracao { get; set; } 
        
        public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
        public DateTime DataAtualizacao { get; set; } = DateTime.UtcNow;
    }
}

File: src/app/models/movie.ts
export interface Movie {
  id?: number;
  titulo: string;
  ano: number;
  genero: string;
  diretor: string;
  sinopse: string;
  posterUrl?: string;
  avaliacao: number; 
  dataLancamento: Date;
  duracao: number;
}

export interface MovieResponse {
  duracao: any;
  avaliacao: any;
  posterUrl: any;
  sinopse: any;
  diretor: any;
  genero: any;
  ano: any;
  titulo: any;
  dataLancamento: string | number | Date;
  movies: Movie[];
  totalCount: number;
  currentPage: number;
  totalPages: number;
}